### ISSUE TITLE
Although the training is well done, nothing is detected. Help me.

### ISSUE THREAD

**Issue by babbu3682** [matterport/Mask_RCNN#1958]
I don't know the answer. Help me!

After successful training, the results of the tensorboard are shown in the figure below. But when I run detect(), there is no output at all. 

My data is medical data for (2017 x 2048) 1 channel. And I have to detect many boundaries like line and segmentation the boundaries there. But nothing is caught in r['class_ids'] or r['masks']. I think the loss is low, so the model has to catch something, but the model doesn't get caught at all. I don’t' know the reason.

------------------------------------------------------------------------------------------------------------------------------------
Weights:  last
Dataset:  /data
Subset:  test
Logs:  /data/volume/logs

Configurations:
BACKBONE                       resnet101
BACKBONE_STRIDES               [4, 8, 16, 32, 64]
BATCH_SIZE                     1
BBOX_STD_DEV                   [0.1 0.1 0.2 0.2]
COMPUTE_BACKBONE_SHAPE         None
DETECTION_MAX_INSTANCES        100
DETECTION_MIN_CONFIDENCE       0
DETECTION_NMS_THRESHOLD        0.3
FPN_CLASSIF_FC_LAYERS_SIZE     1024
GPU_COUNT                      1
GRADIENT_CLIP_NORM             5.0
IMAGES_PER_GPU                 1
IMAGE_CHANNEL_COUNT            1
IMAGE_MAX_DIM                  1024
IMAGE_META_SIZE                21
IMAGE_MIN_DIM                  800
IMAGE_MIN_SCALE                0
IMAGE_RESIZE_MODE              square
IMAGE_SHAPE                    [1024 1024    1]
LEARNING_MOMENTUM              0.9
LEARNING_RATE                  0.001
LOSS_WEIGHTS                   {'rpn_class_loss': 1.0, 'rpn_bbox_loss': 1.0, 'mrcnn_class_loss': 1.0, 'mrcnn_bbox_loss': 1.0, 'mrcnn_mask_loss': 1.0}
MASK_POOL_SIZE                 14
MASK_SHAPE                     [28, 28]
MAX_GT_INSTANCES               100
MEAN_PIXEL                     [123.3]
MINI_MASK_SHAPE                (56, 56)
NAME                           xray
NUM_CLASSES                    9
POOL_SIZE                      7
POST_NMS_ROIS_INFERENCE        1000
POST_NMS_ROIS_TRAINING         2000
PRE_NMS_LIMIT                  6000
ROI_POSITIVE_RATIO             0.33
RPN_ANCHOR_RATIOS              [0.5, 1, 2]
RPN_ANCHOR_SCALES              (32, 64, 128, 256, 512)
RPN_ANCHOR_STRIDE              1
RPN_BBOX_STD_DEV               [0.1 0.1 0.2 0.2]
RPN_NMS_THRESHOLD              0.7
RPN_TRAIN_ANCHORS_PER_IMAGE    256
STEPS_PER_EPOCH                1000
TOP_DOWN_PYRAMID_SIZE          256
TRAIN_BN                       False
TRAIN_ROIS_PER_IMAGE           200
USE_MINI_MASK                  True
USE_RPN_ROIS                   True
VALIDATION_STEPS               50
WEIGHT_DECAY                   0.0001

![스크린샷, 2020-01-18 00-43-22](https://user-images.githubusercontent.com/48959435/72625889-fe20f980-398c-11ea-80df-9d63fe03a575.png)
![스크린샷, 2020-01-18 00-43-12](https://user-images.githubusercontent.com/48959435/72625891-fe20f980-398c-11ea-881b-4976e1dd993f.png)
![스크린샷, 2020-01-18 00-42-48](https://user-images.githubusercontent.com/48959435/72625892-feb99000-398c-11ea-8553-4ea98646db35.png)
![스크린샷, 2020-01-18 00-42-16](https://user-images.githubusercontent.com/48959435/72625894-feb99000-398c-11ea-8b4b-56958c165850.png)
![스크린샷, 2020-01-18 00-41-54](https://user-images.githubusercontent.com/48959435/72625895-ff522680-398c-11ea-82a7-db23dcc8ddee.png)
![1](https://user-images.githubusercontent.com/48959435/72625896-ff522680-398c-11ea-91d9-10242d07147d.png)

**Comment by ThiagoViek** [comment]
I have the same issue. Both train and validation mask_losses go to zero, but r['mask'] is always null. 
I am not using pre-trained weights and my backbone is a MobileNet, though. Bounding box and classification work perfectly after training.

**Comment by babbu3682** [comment]
@ThiagoViek 
Do you also target thin and long objects? I used only a long and thin line as a target object in the image.

**Comment by ThiagoViek** [comment]
Mine as well, but with when I use a ResNet backbone the model works properly and masks are generated. I think I have a problem with the MobileNet backbone.

**Comment by truongtd6285** [comment]
Did you try to run detection on the images that were used to train?

**Comment by jujsd** [comment]
So, excuse me, have you solved the problem yet?           
I encountered the same problem today.  During training, the loss decreased normally. The test showed that the mAP was 0, and I found that r['rois'] was null. 
How do you solve this problem? I checked the input of the test, and there was nothing wrong with it.

**Comment by ghost** [comment]
Hello, was this problem solved?

**Comment by linaouro** [comment]
Try decreasing `DETECTION_MIN_CONFIDENCE`, I trained on a small dataset and most instances where detected with confidence around 80, so when `DETECTION_MIN_CONFIDENCE= 90`, none where found.
